andela/tesla-ah-frontend
# Save Current User after registration and prevent unverified users from creating an article

### What does this PR do?
- Save the current user in the state after successful signup.
- Throw an error if the user who is not verified tries to create an article

### Description
- I discovered a bug that the signup was keeping the user info in the local storage, but  It was changed after the signup feature was merged, and we forgot to modify it. That means the profile was undefined.
- On Create article, we didn't check if the user is verified before allowing them to create articles. What I added is to throw an error if the user is not verified.
### How should it be manually tested
- Clone this repo, switch to this branch
- Run `yarn` to refresh the project dependencies
- Run `yarn dev` to run the project
- Try to create a new account
- After successful registration, try to see if your information are visible ( Click on your Profile Picture)
- Try to create an article if you are not verified yet.
